### ISSUE TITLE
[RFC] Beyond hyperparameter optimization

### ISSUE THREAD

**Issue by MechCoder** [scikit-optimize/scikit-optimize#204]
We should make clear that are not yet another HPO library.

I suggest we should start documenting examples or areas where black-box optimization is applied.

**Comment by betatim** [comment]
Good idea. From the bit of searching for public datasets/objectives I didn't find much. Thinking out loud some ideas for areas where we could look for example problems/datasets or that we could take as basis for a problem:

* aerodynamics (finding best configuration of wings/cars/etc)
* food recipes
* drug design
* compiler flags
* configuring simulators (we might be able to get a dataset here soon)
* [bin packing](https://en.wikipedia.org/wiki/Bin_packing_problem) solutions??

More ideas?

**Comment by betatim** [comment]
* settings for a robot/plane/quadcopter controller

**Comment by MechCoder** [comment]
We can also look for functions that are really expensive to evaluate ;)

**Comment by betatim** [comment]
[SAT solvers](https://en.wikipedia.org/wiki/Boolean_satisfiability_problem#Algorithms_for_solving_SAT) have lots of parameters and are used by a lot of people. Found this by reading SMAC slides.

**Comment by betatim** [comment]
Should we use this issue to try and find a concrete application to help drive the development?
